dexter/src/main/java/com/karumi/dexter/DexterInstance.java

```diff
@@ -89,11 +90,22 @@ void checkPermissions(MultiplePermissionsListener listener, Collection<String> p
   }
 
   /**
+   * Check if there are some permissions pending to be confirmed by the user and restarts the request
+   * for permission process.
+   */
+  void checkPendingPermissions() {
+    boolean shouldContinueRequestingPendingPermissions =
+        !pendingPermissions.isEmpty() && !rationaleShown.get();
+    if (shouldContinueRequestingPendingPermissions) {
+      startTransparentActivity();
```

Comment:
Wouldn't this potentially leak activities? the old activity hasn't been destroyed yet we are creating another one and leaving the old one alive. Given this class is entirely responsible for the transparent activity lifecycle I'd make sure we are only handling one at a time, either closing the old one and relaunching a new one or reusing the original.